### ISSUE TITLE
Add a string endpoint description plugin for the stdio endpoint

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5729]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5729|
|Type|enhancement|
|Created|2012-06-28 13:19:46Z|
|Branch|https://github.com/twisted/twisted/tree/stdio-endpoint-plugin-5729-2|

Following [#4697](https://github.com/twisted/twisted/issues/4697), we need a plugin so that the endpoint can be invoked as something like: serverFromString("stdio").

<details><summary>Searchable metadata</summary>

```
trac-id__5729 5729
type__enhancement enhancement
reporter__ashfall ashfall
priority__normal normal
milestone__ 
branch__branches_stdio_endpoint_plugin_5729_2 branches/stdio-endpoint-plugin-5729-2
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1340889586000000 1340889586000000
changetime__1341846313000000 1341846313000000
version__None None
owner__ashfall ashfall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall commented|
|-|-|

(In [34674]) Branching to 'stdio-endpoint-plugin-5729'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @ashfall|
|-|-|

Thanks!  There are a few conflicts that will need to be resolved in a merged-forward version of this branch.  Otherwise, this looks good to me!  Please merge forward, test the new branch on buildbot, and merge if everything is working.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_4"></a>|@ashfall commented|
|-|-|

(In [34713]) Branching to 'stdio-endpoint-plugin-5729-2'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_5"></a>|@ashfall commented|
|-|-|

[Build Results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/stdio-endpoint-plugin-5729-2)

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_6"></a>|@ashfall set status to `closed`|
|-|-|

(In [34715]) Merge stdio-endpoint-plugin-5729-2: Added a string description plugin for StandardIOEndpoint

Author: ashfall
Reveiwer: exarkun
Fixes: [#5729](https://github.com/twisted/twisted/issues/5729)
